scRNA-seq/04-dimension_reduction_scRNA.Rmd

```diff
@@ -31,226 +42,294 @@ library(magrittr)
+These NAs can be a problem for filtering with a Boolean vector, as we did above, so instead we will use the `which()` function to get the *positions* of the cells that pass our filter and select the columns we want using that.
+
+```{r filter_empty, live = TRUE}
+cells_to_retain <- which(droplet_stats$FDR <= 0.01)
+
+filtered_sce <- hodgkins_sce[, cells_to_retain]
+dim(filtered_sce)
 ```
 
-Looking at the data, it seems like a reasonable cutoff might be 500 detected genes, and a mitochondrial percentage less than 20%
+How does this compare to the number of cells in the Cell Ranger filtered data? 
+Looking the `web_summary.html` report from Cell Ranger, it seems that it would have kept 3,394 cells, so we seem to be getting broadly similar results.
+
+### Checking mitochondrial content
 
-Using these QC data, we can now create a subset of the `SingleCellExperiment` object, keeping only the columns (samples) that pass our QC thresholds:
+While `emptyDrops()` should have filtered out droplets containing no cells, it will not necessarily filter out damaged cells.
+For that we will still want to look at mitochondrial content, as we did previously.  
+The statistics we calculated earlier with `addPerCellQC()` are retained in our new object, so we can plot those directly.
 
-```{r qcfilter, live = TRUE}
-# create tm_sce_filtered object with just the cells that pass filters
-qc_pass <- (tm_sce$detected > 500) & (tm_sce$subsets_mito_percent < 20)
-tm_sce_filtered <- tm_sce[, qc_pass]
+```{r mito_percent_plot}
+# Plot the mitochondrial percents stored in `filtered_sce`
+ggplot(mapping = aes(x = filtered_sce$subsets_mito_percent)) +
+  geom_histogram(bins = 100) 
+```
+There are certainly some cells with high mitochondrial percentages!
+For now, we will use a cutoff of 20% to filter out the worst of the cells.
+
+```{r live = TRUE}
+filtered_sce <- filtered_sce[, filtered_sce$subsets_mito_percent < 20]
 ```
 
-We can also filter by features (genes in our case) using `scater::addPerFeatureQC()` which will computer the number of samples where each gene is detected and the mean count across all genes. 
-We can then use those data (stored in `rowData`) to filter by row to only the genes that are detected in at least 5 cells, and with a mean count > 0.1.
+
+We can also filter by features (genes in our case) using `scater::addPerFeatureQC()` which will compute the number of samples where each gene is detected and the mean count across all genes. 
+We can then use those data (stored in `rowData`) to filter by row to only the genes that are detected in at least 5% of cells, and with a mean count > 0.1.
 
 ```{r gene_qc}
-tm_sce_filtered <- scater::addPerFeatureQC(tm_sce_filtered)
-detected <- rowData(tm_sce_filtered)$detected > 5
-expressed <- rowData(tm_sce_filtered)$mean > 0.1
+filtered_sce <- scater::addPerFeatureQC(filtered_sce)
+detected <- rowData(filtered_sce)$detected > 5
```

Comment:
Hm, I wouldn't necessarily expect this to do what the text says (based on the name) but okay!

Reply:
You mean you wouldn't expect `detected` to be a percent measure? Yeah, neither would I, but that is what it seems to be: 
> detected: numeric, the percentage of observations above threshold. 
https://rdrr.io/bioc/scuttle/man/perFeatureQCMetrics.html